model_api/cpp/models/src/anomaly_model.cpp

```diff
@@ -0,0 +1,230 @@
+  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  See the License for the specific language governing permissions and
+  limitations under the License.
+*/
+
+#include "models/anomaly_model.h"
+
+#include <memory>
+#include <openvino/core/any.hpp>
+#include <openvino/core/model.hpp>
+#include <ostream>
+
+#include "models/image_model.h"
+#include "models/input_data.h"
+#include "models/internal_model_data.h"
+#include "models/results.h"
+#include "utils/slog.hpp"
+
+std::string AnomalyModel::ModelType = "AnomalyDetection";
+
+/// @brief Initializes the model from the given configuration
+/// @param top_priority  Uses this as the primary source for setting the parameters
+/// @param mid_priority Fallback source for setting the parameters
+void AnomalyModel::init_from_config(const ov::AnyMap &top_priority, const ov::AnyMap &mid_priority)
+{
+  mean_values = get_from_any_maps("mean_values", top_priority, mid_priority, mean_values);
+  scale_values = get_from_any_maps("scale_values", top_priority, mid_priority, scale_values);
+  imageShape = get_from_any_maps("image_shape", top_priority, mid_priority, imageShape);
+  imageThreshold = get_from_any_maps("image_threshold", top_priority, mid_priority, imageThreshold);
+  pixelThreshold = get_from_any_maps("pixel_threshold", top_priority, mid_priority, pixelThreshold);
+  normalizationScale = get_from_any_maps("normalization_scale", top_priority, mid_priority, normalizationScale);
+  task = get_from_any_maps("task", top_priority, mid_priority, task);
+  labels = get_from_any_maps("labels", top_priority, mid_priority, labels);
+}
+
+AnomalyModel::AnomalyModel(std::shared_ptr<ov::Model> &model,
+                           const ov::AnyMap &configuration)
+    : ImageModel(model, configuration)
+{
+  init_from_config(configuration, model->get_rt_info<ov::AnyMap>("model_info"));
+}
+
+AnomalyModel::AnomalyModel(std::shared_ptr<InferenceAdapter> &adapter)
+    : ImageModel(adapter)
+{
+  init_from_config(adapter->getModelConfig(), ov::AnyMap{});
+}
+
+std::unique_ptr<AnomalyResult> AnomalyModel::infer(
+    const ImageInputData &inputData)
+{
+  auto result = ModelBase::infer(static_cast<const InputData &>(inputData));
+
+  return std::unique_ptr<AnomalyResult>(
+      static_cast<AnomalyResult *>(result.release()));
+}
+
+std::unique_ptr<ResultBase> AnomalyModel::postprocess(
+    InferenceResult &infResult)
+{
+  ov::Tensor predictions = infResult.outputsData[outputNames[0]];
+  const auto &inputImgSize =
+      infResult.internalModelData->asRef<InternalImageModelData>();
+
+  double pred_score;
+  std::string pred_label;
+  cv::Mat anomaly_map;
+  cv::Mat pred_mask;
+  std::vector<cv::Rect> pred_boxes;
+  if (predictions.get_shape().size() == 1)
+  {
+    pred_score = predictions.data<float>()[0];
+  }
+  else
+  {
+    anomaly_map = cv::Mat(imageShape[0], imageShape[1], CV_32FC1,
+                          predictions.data<float>());
```

Comment:
`anomaly_map` is a `cv::Mat` representation of `predictions`. `predictions` has it's shape, but here you override it with `imageShape`. Please, use `predictions`'s shape. This means you shouldn't have `imageShape` at all